cognite/client/utils/_time.py

```diff
@@ -173,6 +387,222 @@ def split_time_range(start: int, end: int, n_splits: int, granularity_in_ms: int
+        from backports.zoneinfo import ZoneInfo  # type:ignore
+    utc = ZoneInfo("UTC")
+    multiplier, unit = get_granularity_multiplier_and_unit(granularity, standardize=True)
+    if unit in {"h", "m", "s"}:
+        # UTC is always fixed for these intervals
+        return [{"start": start, "end": end, "granularity": f"{multiplier}{unit}"}]
+    start, end = align_large_granularity(start, end, granularity)
+    if unit in VARIABLE_LENGTH_UNITS:
+        return _to_fixed_utc_intervals_variable_unit_length(start, end, multiplier, unit, utc)
+    elif unit in {"day", "week"}:
+        return _to_fixed_utc_intervals_fixed_unit_length(start, end, multiplier, unit, utc)
+    raise ValueError(f"Not supported unit {unit}")
+
+
+def _to_fixed_utc_intervals_variable_unit_length(
+    start: datetime, end: datetime, multiplier: int, unit: str, utc: zoneinfo.ZoneInfo
+) -> list[dict[str, datetime | str]]:
+    pd = cast(Any, local_import("pandas"))
+
+    # Pandas seems to have issues with ZoneInfo object, so removing the timezone and adding it back.
+    index = pd.date_range(start.replace(tzinfo=None), end.replace(tzinfo=None), freq="1D").tz_localize(start.tzinfo.key)  # type: ignore
+    # All units are always using the first of each month, filtering it out here makes index much smaller.
+    index = index[index.day == 1]
+    if unit == "month":
+        month_no = index.month - start.month + 12 * (index.year - start.year)
+        index = index[month_no % multiplier == 0].tz_convert(utc)
+    elif unit == "quarter":
+        quarter_no = (index.month - start.month) // 3 + 4 * (index.year - start.year)
+        index = index[index.month.isin({1, 4, 7, 10}) & (quarter_no % multiplier == 0)].tz_convert(utc)
+    elif unit == "year":
+        year_no = index.year - start.year
+        index = index[(index.month == 1) & (year_no % multiplier == 0)].tz_convert(utc)
+    else:
+        raise ValueError(f"Unit {unit} is not supported.")
+    return [
+        {
+            "start": start.to_pydatetime(),
+            "end": end.to_pydatetime(),
+            "granularity": f"{(end-start).total_seconds()//3600:.0f}h",
+        }
+        for start, end in zip(index[:-1], index[1:])
+    ]
+
+
+def _to_fixed_utc_intervals_fixed_unit_length(
+    start: datetime, end: datetime, multiplier: int, unit: str, utc: zoneinfo.ZoneInfo
+) -> list[dict[str, datetime | str]]:
+    pd = cast(Any, local_import("pandas"))
+
+    freq = granularity_in_hours(multiplier, unit)
+    # Pandas seems to have issues with ZoneInfo object, so removing the timezone and adding it back.
+    index = pd.date_range(start.replace(tzinfo=None), end.replace(tzinfo=None), freq=f"{freq}H").tz_localize(
+        start.tzinfo.key  # type: ignore
+    )
+    expected_freq = pd.Timedelta(f"{freq}H")
+    next_diff = index - index.to_series().shift(1)
+    last_diff = index - index.to_series().shift(-1)
+    index = index[(last_diff.abs() != expected_freq) | (next_diff.abs() != expected_freq)]
```

Comment:
Perhaps swap subtraction on `last_` and get rid of all `abs` calls. 

Since calling `to_series()` on an index doubles the amount of data I would also suggest you only store the "mask" with boolean values to a variable:
```suggestion
    next_mask = (index - index.to_series().shift(1)) != expected_freq
    last_mask = (index.to_series().shift(-1) - index) != expected_freq
    index = index[last_mask | next_mask]
```